File: sdk/python/hitechcloud/resources/vcloudstack.py
"""vCloudStack Public Cloud resource (4 endpoints)"""

from .base import BaseResource

class VCloudStackResource(BaseResource):
    """vCloudStack Public Cloud: rescue, unrescue, console, usage"""

    def rescue_vm(self, service_id: int, vmid: int) -> dict:
        """POST /api/service/{id}/vms/{vmid}/rescue - Reboot VM into rescue mode"""
        return self._http.post(f"/api/service/{service_id}/vms/{vmid}/rescue")

    def unrescue_vm(self, service_id: int, vmid: int) -> dict:
        """POST /api/service/{id}/vms/{vmid}/unrescue - Disable rescue mode"""
        return self._http.post(f"/api/service/{service_id}/vms/{vmid}/unrescue")

    def get_console(self, service_id: int, vmid: int) -> dict:
        """GET /api/service/{id}/vms/{vmid}/console - Get VM console"""
        return self._http.get(f"/api/service/{service_id}/vms/{vmid}/console")

    def get_usage(self, service_id: int, vmid: int) -> dict:
        """GET /api/service/{id}/vms/{vmid}/usage - Get VM usage details"""
        return self._http.get(f"/api/service/{service_id}/vms/{vmid}/usage")

File: sdk/ruby/lib/hitechcloud/resources/vcloudstack.rb
module HiTechCloud
  module Resources
    class Vcloudstack < Base

        # Enter rescue
        def rescue(id, vmid)
          @http.post("/api/service/%{id}/vms/%{vmid}/rescue" % { id: id, vmid: vmid })
        end

        # Exit rescue
        def unrescue(id, vmid)
          @http.post("/api/service/%{id}/vms/%{vmid}/unrescue" % { id: id, vmid: vmid })
        end

        # Get console URL
        def console(id, vmid)
          @http.get("/api/service/%{id}/vms/%{vmid}/console" % { id: id, vmid: vmid })
        end

        # Get usage
        def usage(id, vmid)
          @http.get("/api/service/%{id}/vms/%{vmid}/usage" % { id: id, vmid: vmid })
      end
    end
  end
end
